tests/receivers/smartagent/collectd-kafka/bundled_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package tests
+
+import (
+	"fmt"
+	"path"
+	"runtime"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+var kafka = testutils.NewContainer().WithContext(
+	path.Join(".", "testdata", "kafka"),
+).WithEnv(map[string]string{
+	"KAFKA_ZOOKEEPER_CONNECT": "zookeeper:2181",
+}).WithNetworks("kafka")
+
+var kafkaZookeeper = testutils.NewContainer().WithImage("zookeeper:3.5").WithName("zookeeper").WithNetworks("kafka").WithExposedPorts("2181:2181").WillWaitForPorts("2181")
+
+func TestKafkaBrokerMetrics(t *testing.T) {
+	testutils.SkipIfNotContainerTest(t)
+	if runtime.GOOS == "darwin" {
+		t.Skip("unable to share sockets between mac and d4m vm: https://github.com/docker/for-mac/issues/483#issuecomment-758836836")
+	}
+	tc := testutils.NewTestcase(t)
+	defer tc.PrintLogsOnFailure()
+	defer tc.ShutdownOTLPReceiverSink()
+
+	kafkaBroker := kafka.WithName("prefix-kafka-test-broker-suffix").WithEnvVar("START_AS", "broker").WithExposedPorts("7099:7099", "9092:9092").WillWaitForPorts("7099", "9092").WillWaitForLogs("INFO Opening socket connection to server")
```

Comment:
These can be defined as globals, and to keep test overhead down it may be preferable to start shared instances like is done w/ the existing tests.